### PR TITLE
rm: rm3.sh now passes

### PR DESCRIPTION
**Pull request by palaster**
- Changed the use of hard-coded prompts messages to use uucore::util_name instead
- Rewrote prompt_write_protected and prompt_file into a single function prompt_file because I a lot of the code was being reused for both

### CODE REVIEW

File: src/uu/rm/src/rm.rs [hunk 1]
```diff
@@ -250,14 +248,38 @@ fn remove(files: &[String], options: &Options) -> bool {
...
-                    handle_dir(file, options)
-                } else if is_symlink_dir(&metadata) {
+            Ok(symlink_metadata) => {
+                if is_symlink_dir(&symlink_metadata) {
                     remove_dir(file, options)
-                } else {
+                } else if symlink_metadata.is_symlink() {
                     remove_file(file, options)
+                } else {
+                    match file.metadata() {
```
Comment (sylvestre) [inline src/uu/rm/src/rm.rs:259]: please add a comment explaining in which case we are here

File: src/uu/rm/src/rm.rs [hunk 2]
```diff
@@ -250,14 +248,38 @@ fn remove(files: &[String], options: &Options) -> bool {
...
+                        Ok(metadata) => {
+                            if metadata.is_dir() {
+                                handle_dir(file, options)
+                            } else {
+                                remove_file(file, options)
+                            }
+                        }
+                        Err(_e) => {
```
Comment (sylvestre) [inline src/uu/rm/src/rm.rs:267]: please add a test to cover this case

**Review by sylvestre — CHANGES_REQUESTED** [review]
please add a test in the rust test suite

File: src/uu/rm/src/rm.rs (hunk 2 above)
Comment (palaster) [reply src/uu/rm/src/rm.rs:267]: There is already a test for it: test_rm::test_rm_symlink_dir() and test_rm::test_rm_invalid_symlink()

Comment (sylvestre) [reply src/uu/rm/src/rm.rs:267]: the code coverage results don't agree ;)

**Comment by palaster** [comment]
Broke branch will have to make a new one

**Pull request closed without merging**